src/main/java/org/opensearch/knn/index/KNNSettings.java

```diff
@@ -326,7 +332,7 @@ private void setSettingsUpdateConsumers() {
             );
 
             NativeMemoryCacheManager.getInstance().rebuildCache(builder.build());
-        }, new ArrayList<>(dynamicCacheSettings.values()));
+        }, Stream.concat(dynamicCacheSettings.values().stream(), FEATURE_FLAGS.values().stream()).collect(Collectors.toList()));
```

Comment:
can we make this list immutable , since we are not anticipating , any caller to mutate future flag and that ensures that this is also not getting mutated by mistake.